### ISSUE TITLE
Conditionally or optionally hidden Fieldset

### ISSUE THREAD

**Issue by eghernqvist** [sanity-io/sanity#2763]
**Is your feature request related to a problem? Please describe.**
We want to have fieldsets which consist of conditional fields, hidden based on a sibling field. The fieldset, even if empty, still displays it's title, which causes confusing UI.

**Describe the solution you'd like**
I would like to be able to conditionally hide whole fieldsets, something like this:
```
      fieldsets: [
        {
          title: "Date span",
          name: "datespan",
          hidden: ({ document }) => !document?.types?.includes('versionNumbers'),
        },
      ],
```
Where `types` is an array of strings used for conditional fields within the `datespan` fieldset.

**Describe alternatives you've considered**
Another alternative would be to have an option on fieldsets which cause them to hide automatically when empty, somehow. Something like this:
```
      fieldsets: [
        {
          title: "Date span",
          name: "datespan",
          options: { autohide: true }
        },
      ],
```

**Additional context**
![2021-09-10_13-11-03-621](https://user-images.githubusercontent.com/10044687/132844916-838bf93c-0275-46a6-8f27-5394a8896624.gif)

**Comment by vin-ni** [comment]
+1 :)

**Comment by miracle2k** [comment]
Ran into the same issue. Expected the fieldset UI to automatically disappear if all its fields are hidden.

**Comment by jamesmallred** [comment]
A useful workaround is to create objects and conditionally hide the object based on the 'Types' value (or sibling field value). There is an option to make objects collapsible and set it as closed by default.

[Object Type](https://www.sanity.io/docs/object-type)
[Conditional Fields](https://www.sanity.io/docs/conditional-fields)

**Comment by geball** [comment]
Resolved in v2.23.0.
